packages/itwinui-css/src/input/input-flex-container.scss

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@mixin iui-input-flex-container {
+  --_iui-input-background-color: var(--iui-color-background);
+  --_iui-input-border-color: var(--iui-color-border-foreground);
+  --_iui-input-accent-border-color: var(--iui-color-border-accent);
+  --_iui-input-text-color: var(--iui-color-text);
+
+  display: inline-flex;
+  align-items: center;
+  background-color: var(--_iui-input-background-color);
+  border-radius: var(--iui-border-radius-1);
+  color: var(--_iui-input-text-color);
+  position: relative;
+  height: var(--_iui-input-min-height);
+  width: 100%;
+
+  // Adding before pseudo-element to make inner elements more centered
+  &::before {
+    content: '';
+    position: absolute;
+    inset: 0;
+    border: 1px solid var(--_iui-input-border-color);
+    border-radius: var(--iui-border-radius-1);
+  }
+
+  &:has(input:focus-visible) {
+    outline: 2px solid var(--_iui-input-accent-border-color);
+    outline-offset: -2px;
+  }
+
+  @supports not selector(:has(+ *)) {
+    &:where(:focus-within) {
+      outline: 2px solid var(--_iui-input-accent-border-color);
+      outline-offset: -2px;
+    }
+  }
+
+  input {
+    all: unset;
+    z-index: 2;
+    min-width: var(--_iui-input-min-height);
+    flex: 999;
+    padding-inline-start: var(--iui-size-s);
+
+    &:is(:not(:first-child)) {
+      padding-inline-start: var(--iui-size-2xs);
```

Comment:
what's the thinking behind modifying padding if not first child?

Reply:
It seemed a bit too big gap between button and input in not adjusted version:
![image](https://user-images.githubusercontent.com/83585998/235941891-207b5154-4601-4c83-9ba7-771c09edcac8.png)